src/main.cpp

```diff
@@ -397,16 +397,16 @@ static std::optional<struct timespec> get_boottime()
   if (!include_files.empty() && driver.root_->c_definitions.empty())
     driver.root_->c_definitions = "#define __BPFTRACE_DUMMY__";
 
-  if (!clang.parse(driver.root_, bpftrace, extra_flags))
+  if (!clang.parse(driver.root_.get(), bpftrace, extra_flags))
     return nullptr;
 
   err = driver.parse();
   if (err)
     return nullptr;
 
-  auto ast = driver.root_;
-  driver.root_ = nullptr;
-  return std::unique_ptr<ast::Node>(ast);
+  std::unique_ptr<ast::Node> retval = std::move(driver.root_);
+  driver.root_.reset();
+  return retval;
```

Comment:
I feel this could be just `return (ast::Node *)driver.root_.release();`  ?

Reply:
Yup, this can be simplified. Changing it to `return std::move(driver.root_)`.